### ISSUE TITLE
Can't connect to API behind basic auth

### ISSUE THREAD

**Issue by worldworm** [netdata/netdata#3548]
Hi there.

Iam running netdata behind a basic auth reverse proxy (apache2).
Everything else is working great.

---

When including the dashboard.js file inside a html file, and i open this file:
My browser (firefox) asks for basic auth, after inserting its loading all js files (jquery, dashboard, bootstrap ...).
After a few seconds the data for the gauge are getting loaded (".../api/v1/data?chart=netdata.server_cpu.."). The problem here is, these requests stops with a 401 error (Unauthorized)

![grafik](https://user-images.githubusercontent.com/13227454/37373338-76ee4240-2716-11e8-9698-ece1fefd2981.png)

The gauges are responding with a "data download failed"

---

When i insert the user and pw in the path (not great, but it works)
For example https://user:pw@exampe.com/dashboard.js
The js file and also the others( jquery, dashboard, bootstrap ...) are also loaded via this auth methode (so no password auth window popup).
But the gauges and any other data isn't loaded. Console.log = "ERROR 404: Chart not found"
Also there are no data requests in the Networklog, so something is failing here on client side i think.

**Comment by ktsaou** [comment]
Late reply: did you solve it?

I am closing this, but feel free to post if the issue remains.

**Comment by cakrit** [comment]
@metaphys which version of netdata do you have? Are you also using apache2 and Firefox?

**Comment by metaphys** [comment]
Yeah thanks for reopening this one.
Firefox 60.4.0esr
Nginx 1.12.2 with auth_basic
netdata: 1.10.0-849-gcaae263_rolling
My situation is a bit different but the connection to api fails with error 404 when connecting with credentials given in the data-host url.
1) I have a pool of virtual host, on each of them runs nginx with ssl certs (https)
each vm has its own netdata installation running on internal IP:19999 and nginx on the host is proxing url like https://myHost.myDomain/myVM/netdata to the correct VM port 19999. This is working perfectly. I go to the url, i'm asked for credential and I have access to netdata

2) Now on one of the host, I added a custom dashboard to see all the VMs and hosts status. First I had to fight with CORS but I got it out of the way. 
I tested this without authentication and it does work.

3) Finally I activated Authentication. Since I'm no webdev guru, I don't know any other way to deal with the authentication than adding the credential to the url like @worldworm I assume it's ok since the access is firewalled, ssl encrypted, password protected.
When I connect to the custom dashboard page, I'm asked for credential and all netdata's information VMs and the host on the which one is the  custom dashboard's file are displayed ok. All other charts relative to others host and their VMs are failing with '... chart not found on url ...'
If I go to firefox dev tools console, I have list of
ERROR 404: Chart not found: /api/v1/chart?chart=system.cpu
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.ram
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.io
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.net
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.cpu
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.ram
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.io
dashboard.js:1017:9
ERROR 404: Chart not found: /api/v1/chart?chart=system.net
dashboard.js:1017:9
